### PR TITLE
Add optional tokio implementations

### PR DESCRIPTION
**Pull request by lovesegfault**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,3 +15,4 @@ documentation = "https://docs.rs/sendfd"
 
 [dependencies]
 libc = "0.2"
+tokio = { version = "0.2", features = [ "uds" ], optional = true }
```
Comment (nagisa) [inline Cargo.toml]: May want to name this in some way that makes it obvious this is 0.2, given that 0.3 is now the "current".

**Review by nagisa — APPROVED** [review]

File: src/lib.rs [hunk 2]
```diff
@@ -4,6 +4,12 @@ use std::os::unix::io::{AsRawFd, RawFd};
 use std::os::unix::net;
 use std::{alloc, io, mem, ptr};
 
+#[cfg(feature = "tokio_0_2")]
+extern crate tokio_0_2 as tokio;
+
+#[cfg(feature = "tokio_0_3")]
+extern crate tokio_0_3 as tokio;
```
Comment (nagisa) [inline src/lib.rs]: renaming both `as tokio` makes features not additive due to conflicting names. With both enabled you should end up with impls for both tokio 0.2 and 0.3.

Comment (lovesegfault) [reply src/lib.rs]: Right, the alternative is to duplicate the implementation code, do we want that?

Comment (nagisa) [reply src/lib.rs]: It is already fairly deduplicated by the virtue of those impls just calling a single function, isn't it? A macro could reduce LOC further. Overall I don't think saving on LOC in exchange for surprising behaviour of the features is worth it.

Comment (lovesegfault) [reply src/lib.rs]: Done

**Review by nagisa — APPROVED** [review]

**Pull request merged**